api/v1beta1/nodemaintenance_webhook.go

```diff
@@ -160,6 +165,25 @@ func (v *NodeMaintenanceValidator) validateNoNodeMaintenanceExists(nodeName stri
 	return nil
 }
 
+func (v *NodeMaintenanceValidator) validateNodeIsReady(nodeName string) error {
+	node, err := getNode(nodeName, v.client)
+	if err != nil {
+		return fmt.Errorf("could not get node for validating node is Ready, please try again: %v", err)
+	} else if node == nil {
+		return fmt.Errorf(ErrorNodeNotExists, nodeName)
+	}
+	// find the status of node condition Ready
+	nodeConditionByType := make(map[corev1.NodeConditionType]corev1.NodeCondition)
+	for _, nc := range node.Status.Conditions {
+		nodeConditionByType[nc.Type] = nc
```

Comment:
what's the value of using the `nodeConditionByType` map, and not directly checking for the `Ready` condition here? 🤔

Reply:
I thought it looked nicer, but I see your point.